### PR TITLE
Add `TemplateTranslation.variables`

### PR DESCRIPTION
**Pull request by rowanseymour**
Ok we got it right this time ?

Note that this change just adds stuff.. 100% backward compatible

### CODE REVIEW

File: temba/templates/tests.py [hunk 1]
```diff
@@ -739,7 +821,7 @@ def test_parse_language(self):
         self.assertEqual("eng-US", parse_language("en_US"))
         self.assertEqual("fil", parse_language("fil"))
 
-    def test_extract_params(self):
-        self.assertEqual([{"type": "text"}, {"type": "text"}], _extract_params("Hi {{1}} how are you? {{2}}"))
-        self.assertEqual([{"type": "text"}, {"type": "text"}], _extract_params("Hi {{1}} how are you? {{2}} {{1}}"))
-        self.assertEqual([], _extract_params("Hi there."))
+    def test_extract_variables(self):
+        self.assertEqual(["1", "2"], _extract_variables("Hi {{2}} how are you? {{ 1 }}"))
```
Comment (rowanseymour) [inline temba/templates/tests.py:825]: I think it's Twilio's docs where they have some examples that have spaces around the variable name

File: temba/api/internal/tests.py [hunk 2]
```diff
@@ -93,7 +93,16 @@ def test_templates(self):
...
+            components=[
+                {
+                    "type": "body",
+                    "name": "body",
+                    "content": "Hi {{1}}",
+                    "variables": {"1": 0},
+                    "params": [{"type": "text"}],
+                }
+            ],
+            variables=[{"type": "text"}],
```
Comment (rowanseymour) [inline temba/api/internal/tests.py:105]: using objects just in case

File: temba/templates/whatsapp.py [hunk 3]
```diff
@@ -13,51 +13,74 @@
...
-    extracted = []
-    all_supported = True
+    components = []
+    variables = []
+    supported = True
+
+    def add_variables(names: list, typ: str) -> dict:
+        map = {}
+        for name in names:
+            variables.append({"type": typ})
```
Comment (norkans7) [inline temba/templates/whatsapp.py:35]: This works for the currently WA templates but we need a way to make sure for Twilio content that the shared variable in multiple components is added once.

I am wondering whether we can move the `extract_component` to be a method on the channel type or have a way to override that for some channel types

Comment (rowanseymour) [reply temba/templates/whatsapp.py:35]: yeah this is the function to turn WhatsApp templates into our format - I'm envisaging channel types being able to specify a template type `(whatsapp|twilio)`

Comment (norkans7) [reply temba/templates/whatsapp.py:35]: So we pass `(whatsapp|twilio)` to `extract_components ` which will pass that to `add_variables`?

Comment (norkans7) [reply temba/templates/whatsapp.py:35]: Or we add a `twilio.py` that will contains its own approach to extract the components?

Comment (rowanseymour) [reply temba/templates/whatsapp.py:35]: yeah something like that

**Review by norkans7 — APPROVED** [review]

**Pull request merged**